### PR TITLE
Add Docker build / run instructions

### PR DESCRIPTION
**Pull request by jnewland**
Adds a Dockerfile and documents how to use it. 

tl;dr:

```bash
~/src/appdaemon (docker) 
$ docker build -t appdaemon .
Sending build context to Docker daemon 50.18 kB
Step 1 : FROM python:3.4
 ---> 93bc8e41eb8c
Step 2 : RUN mkdir -p /usr/src/app
 ---> Using cache
 ---> bdb71898d80e
Step 3 : WORKDIR /usr/src/app
 ---> Using cache
 ---> 5817bdb6fa26
Step 4 : VOLUME /conf
 ---> Using cache
 ---> f850f982963f
Step 5 : RUN pip3 install --no-cache-dir   astral   configparser   daemonize   sseclient
 ---> Using cache
 ---> 5da7377daf33
Step 6 : COPY . .
 ---> Using cache
 ---> d1af06f395c1
Step 7 : CMD /usr/local/bin/python /usr/src/app/bin/appdaemon.py /conf/appdaemon.cfg
 ---> Using cache
 ---> c7c207563b2b
Successfully built c7c207563b2b
~/src/appdaemon (docker) 
$ ls /Users/jnewland/src/ha-config/appdaemon.cfg /Users/jnewland/src/ha-config/apps/*.py
/Users/jnewland/src/ha-config/appdaemon.cfg	/Users/jnewland/src/ha-config/apps/magic.py
~/src/appdaemon (docker) 
$ docker run -d -v /Users/jnewland/src/ha-config:/conf --name appdaemon appdaemon:latest
4a069bb9ad16e367617158a9ec04232e7e141541ba3d330156a163834e6b6330
~/src/appdaemon (docker) 
$ docker logs -ft appdaemon
2016-08-21T14:33:01.360162177Z 2016-08-21 14:33:01,359 INFO Got initial state
2016-08-21T14:33:01.364242752Z 2016-08-21 14:33:01,363 INFO Loading Module: /conf/apps/magic.py
^C
~/src/appdaemon (docker) 
$ docker stop appdaemon
appdaemon
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+RUN pip3 install --no-cache-dir \
+  astral \
+  configparser \
+  daemonize \
+  sseclient
+
+# TODO
+# COPY requirements.txt requirements.txt
+# RUN pip3 install --no-cache-dir -r requirements.txt
```
Comment (jnewland) [inline Dockerfile]: /cc https://github.com/acockburn/appdaemon/issues/3

**Comment by acockburn** [comment]
Thanks for this - you seem to be relying on functionality that isn't yet in place, namely pip3 installation - I do plan on adding this but its going to make merging this a lot harder ... I already have a Docker in testing for the old install method and will probably just update that to use pip3 when I have it done. If so I'll close this without merging it but I will refer to it and we will end up with Docker support in any case.

**Comment by jnewland** [comment]
> Thanks for this - you seem to be relying on functionality that isn't yet in place, namely pip3 installation - I do plan on adding this but its going to make merging this a lot harder ... I already have a Docker in testing for the old install method and will probably just update that to use pip3 when I have it done.

I'm not relying on it at all. The section I commented on above has it commented out. What's actually in use is this:

```
RUN pip3 install --no-cache-dir \
  astral \
  configparser \
  daemonize \
  sseclient
```

I'd be happy to remove the commented out section, or do whatever necessary to make this easier to merge. I'm also happy to just wait on your version. Just let me know what you'd prefer!

**Comment by acockburn** [comment]
Ahh apologies, I missed that!

I am looking at it right now - I have a Dockerfile but it is mysteriously not working so if yours does I'll see if I can figure out the difference, however I think I'll add the PIP support in first to make everything easier - watch this space!

**Comment by acockburn** [comment]
I also like the way you have structured the files and apps along with HA - I will definitely adopt that and change the manual instructions to match and perhaps even make them defaults so the config file for Docker will be even simpler.

**Comment by acockburn** [comment]
Closing this - I didn't merge the PR, I merged it manually and used it as a basis - thanks!

**Comment by jnewland** [comment]
Cool! I don't see a Dockerfile in https://github.com/acockburn/appdaemon. Did you push what you merged manually?

**Comment by acockburn** [comment]
Not yet - I am adding a few other bits and pieces before I push the new version

**Pull request closed without merging**